machine-runner/src/design/protocol.ts

```diff
@@ -21,14 +15,13 @@ import { ReadonlyNonZeroTuple } from '../utils/type-utils.js'
  */
 export type SwarmProtocol<
   SwarmProtocolName extends string,
-  RegisteredEventsFactoriesTuple extends MachineEvent.Factory.ReadonlyNonZeroTuple,
+  MachineEventFactories extends MachineEvent.Factory.Any,
```

Comment:
The name suggests that these are factories; extending from `Factory.Any` says that is should be a union of factories, but if I read it correctly, then it is actually a union of the payloads. Is this correct?

Reply:
No, it is union of factories.
`ReturnType<TheFactories["make"]>` would be the union of events.